File: Cpp/day_1.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::string filename = R"(..\inputs\day_1.txt)";

    int dial = 50;
    int zeros = 0;

    std::ifstream file(filename);
    if (!file.is_open()) {
        std::cerr << "Error opening file!" << std::endl;
        return 1;
    }

    std::string line;
    while (std::getline(file, line)) {
        char direction = line[0];
        int value = std::stoi(line.substr(1));

        if (direction == 'R') {
            dial += value;
        } else {
            dial -= value;
        }

        while (dial > 99) {
            dial -= 100;
        }

        while (dial < 0) {
            dial += 100;
        }


        if (dial == 0) {
            zeros++;
        }
    }

    file.close();
    std::cout << zeros << std::endl;

    return 0;
}

File: Python/day_1.py
x = "../inputs/day_1.txt"

dial: int = 50  # dial starts at 50
zeros: int = 0  # counting the number of times the dial returns to 0

with open(x, "r", encoding="utf-8") as f:  # reading the input
    for line in f:
        line = line.strip()
        direction = line[0]  # 'L' / 'R'
        value = int(line[1:])  # number part
        # print(direction, value) # debug

        if direction == "R":
            dial += value
        else:
            dial -= value

        # tuning for under / over values
        while dial > 99:
            dial -= 100

        while dial < 0:
            dial += 100

        # final check
        if dial == 0:
            zeros += 1
print(zeros)
